rcl/src/rcl/validate_security_context_name.c

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl/validate_security_context_name.h"
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <rcutils/snprintf.h>
+
+#include "rmw/validate_namespace.h"
+
+#include "rcl/error_handling.h"
+
+#include "./common.h"
+
+rcl_ret_t
+rcl_validate_security_context_name(
+  const char * security_context,
+  int * validation_result,
+  size_t * invalid_index)
+{
+  if (!security_context) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  return rmw_validate_namespace_with_size(
+    security_context, strlen(security_context), validation_result, invalid_index);
+}
+
+rcl_ret_t
+rcl_validate_security_context_name_with_size(
+  const char * security_context,
+  size_t security_context_length,
+  int * validation_result,
+  size_t * invalid_index)
+{
+  if (!security_context) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  if (!validation_result) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+
+  int t_validation_result;
```

Comment:
@ivanpauno what does the `t_` stand for? Why not: 
```suggestion
  int rmw_validation_result;
```
?

Same elsewhere.

Reply:
temporal? to differentiate from the parameter being passed.

Reply:
Consider using `tmp`, it's somewhat more common.

Reply:
See https://github.com/ros2/rcl/pull/515/commits/44a15be05484088c78a4e7cde08a021356757160.